packages/core/src/lib/Select/SelectBeta/SelectBeta.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useState, useRef, useEffect } from 'react';
+import { ISelectProps, KEY, OptionProps } from '../types';
+import styles from './SelectBeta.styles';
+import SelectMenuList from './components/SelectMenuList';
+import SelectedItemsList from './components/SelectedItemsList';
+
+const {
+    ButtonStyledSelect,
+    DivStyledDesc,
+    DivStyledOverlay,
+    DivStyledSelectWrapper,
+    DivStyledWrapper,
+    LabelStyled,
+    TriangleDownIconStyled,
+    TriangleUpIconStyled,
+} = styles;
+
+const SelectBeta: React.FunctionComponent<ISelectProps> = ({
+    defaultOptionIndex,
+    disabled = false,
+    id,
+    isMulti = false,
+    isSearch = true,
+    label = 'Select',
+    max = Infinity,
+    name = 'options',
+    onChange,
+    options = [],
+    placeholder,
+    style,
+    value = [],
+    width = '100%',
+    ...props
+}) => {
+    const {
+        customNoDataText,
+        description,
+        errorMessage,
+        onBlurTraditional,
+        onChangeTraditional,
+        prefixIcon,
+        prefixText,
+        refTraditional,
+        state,
+        traditionalHTML5,
+        validation,
+        ...rest
+    } = props;
+    const triggerRef = useRef<HTMLButtonElement>(null);
+    const [isOpen, setIsOpen] = useState(false);
+
+    function elId(component: string) {
+        if (component === undefined) return;
+        // Replace all space with '-' : to fix no scroll issue
+        const newName = name.replace(/ /g, '-');
+        return `w3uik-${newName}-${component}`;
+    }
+
+    function addItem(option: string | undefined) {
+        if (!option) return;
+        // multi selection
+        if (isMulti === true) {
+            let selected = [...value];
```

Comment:
i think this can be a const, right? you are making an array to mutate so i do not think you need a `let`

Reply:
aah right nice catch 🙌🔥

Reply:
This is fixed ✅